src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -647,42 +648,182 @@ public void Fault(RequestChannel channel)
                             throw;
                         }
                     }
+                }
 
-                    try
+                private bool PrepareMessageHeaders(Message message)
```

Comment:
It's not feasible to match up the red to the green as the original code was like spaghetti and took a lot of effort to unravel. If a step has been missed out or is in the wrong order, we should pick it up in our tests as the requests need to be very precisely structured and if something is out of place, the service will reject it.  
I would focus on the green code and check for error handling, confusing code where a clarifying comment would be helpful, and bad code practices such as not guarding for null's etc.